core/src/main/java/org/apache/iceberg/encryption/StandardEncryptionManager.java

```diff
@@ -102,106 +134,69 @@ private SecureRandom workerRNG() {
   }
 
   public ByteBuffer wrapKey(ByteBuffer secretKey) {
-    if (kmsClient == null) {
+    if (keyData == null) {
       throw new IllegalStateException(
           "Cannot wrap key after called after serialization (missing KMS client)");
     }
 
-    return kmsClient.wrapKey(secretKey, tableKeyId);
+    return keyData.kmsClient.wrapKey(secretKey, tableKeyId);
   }
 
   public ByteBuffer unwrapKey(ByteBuffer wrappedSecretKey) {
-    if (kmsClient == null) {
+    if (keyData == null) {
       throw new IllegalStateException("Cannot unwrap key after serialization (missing KMS client)");
     }
 
-    return kmsClient.unwrapKey(wrappedSecretKey, tableKeyId);
+    return keyData.kmsClient.unwrapKey(wrappedSecretKey, tableKeyId);
   }
 
-  void addKekCache(Map<String, KeyEncryptionKey> wrappedKekCache) {
-    for (Map.Entry<String, KeyEncryptionKey> entry : wrappedKekCache.entrySet()) {
-      KeyEncryptionKey wrappedKek = entry.getValue();
-      KeyEncryptionKey cachedKek = kekCache.get(entry.getKey());
-
-      if (cachedKek != null) {
-        Preconditions.checkState(
-            cachedKek.wrappedKey().equals(wrappedKek.wrappedKey()),
-            "Cached kek wrap differs from newly added for %s",
-            entry.getKey());
-      } else {
-        ByteBuffer encryptedKEK =
-            ByteBuffer.wrap(Base64.getDecoder().decode(wrappedKek.wrappedKey()));
-        // Unwrap the key in KMS
-        byte[] kekBytes = unwrapKey(encryptedKEK).array();
-
-        kekCache.put(
-            entry.getKey(),
-            new KeyEncryptionKey(
-                wrappedKek.id(), kekBytes, wrappedKek.wrappedKey(), wrappedKek.timestamp()));
-      }
+  public String currentSnapshotKeyId() {
```

Comment:
I think that manifest list encryption only requires `currentSnapshotKeyId()` and `unwrapKey(id)`. I don't think we need the `wrapKey(ByteBuffer)` and `unwrapKey(ByteBuffer)` methods anymore. We should deprecate them unless we find that we do need them in the next steps.